src/Sparrow/Json/Parsing/JsonParserState.cs

```diff
@@ -104,10 +116,33 @@ public static void FindEscapePositionsIn(FastList<int> buffer, byte* str, int le
                 {
                     buffer.Add(i - lastEscape);
                     lastEscape = i + 1;
+                    continue;
+                }
+
+                //Control character ascii values
+                if (value < 32 || value >= 127 && value <= 159)
+                {
+                    // move rest of buffer 
+                    // write \u0000
+                    // update size
+                    var from = str + i + 1;
+                    var to = str + i + 1 + SizeOfUtf8AsStringInBytesWithoutEscaping;
+                    var sizeToCopy = len - i;
+                    //here we only shifting by 5 bytes since we are going to override the byte at the current position.
+                    Memory.Copy(to, from, sizeToCopy ); //TODO: we must verify that we have enough space in our buffer otherwise we must request a bigger one
```

Comment:
We shouldn't request more, but should validate that we aren't over the max size

Reply:
@talweiss1982 that todo remark is very important! I think that we may want to perform that modification before that function, moreover, we should first count that amount of "control charachters", according to that, to allocate or reallocate the right memory chunk and only then to perform that code

Reply:
So after talking with @talweiss1982 , it seems that I've overlooked the modifications to FindEscapePositionsMaxSize, It wtill seems a bit awkward to modify data inside FindEscapePositionsIn, but as an intermidiate solution, it does the work